landlab/components/weathering/exponential_weathering.py

```diff
@@ -133,8 +133,12 @@ def __init__(
     def calc_soil_prod_rate(self):
         """Calculate soil production rate."""
         # apply exponential function
-        self._soil_prod_rate[self._grid.core_nodes] = self._w0 * np.exp(
-            -self._depth[self._grid.core_nodes] / self._wstar
+        w0_vector = np.broadcast_to(self._w0, self._depth.shape)
+        wstar_vector = np.broadcast_to(self._wstar, self._depth.shape)
```

Comment:
@BCampforts What do you think of doing this broadcasting in `__init__`?

i.e.
```python
        self._wstar = np.broadcast_to(soil_production__decay_depth, grid.number_of_nodes)
        self._w0 = np.broadcast_to(soil_production__maximum_rate, grid.number_of_nodes)
```

Reply:
You should also add a check that *soil_production__decay_depth* is greater than zero.

Reply:
I moved the broadcasting to init. Small disadvantage is that you get an array from the getter in case you set a scalar. Also checked for negative values. Added a tests for that in the init as well.